src/mon/AuthMonitor.cc

```diff
@@ -1933,6 +1934,46 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
   return false;
 }
 
+
+int AuthMonitor::_is_perm_in_mdscaps_correct(const string& mdscaps,
+					     ostream& out)
+{
+  unsigned int num_of_caps = 1;
+  for (auto j : mdscaps) {
+    if (j == ',') {
+      ++num_of_caps;
+    }
+  }
+
+  unsigned int beg = 0;
+  unsigned int len = num_of_caps ? mdscaps.size() : mdscaps.find(",");
```

Comment:
num_of_caps is always going to be >=1, therefore this code will always assign `mdscaps.size()` to `len`. looks a bit buggy

Reply:
Yeah, I miswrote this condition. Fixed now.